### ISSUE TITLE
It does not work at all

### ISSUE THREAD

**Issue by kopyl** [IDEA-Research/GroundingDINO#39]
My command:

```
CUDA_VISIBLE_DEVICES=0 python /workspace/GroundingDINO/demo/inference_on_a_image.py \
  -c /workspace/GroundingDINO/groundingdino/config/GroundingDINO_SwinB.cfg.py \
  -p /workspace/groundingdino_swinb_cogcoor.pth \
  -i /workspace/1e72fd7-1hordon-690.png \
  -o "outputs/0" \
  -t "cat ear." 
```

Input image:
![image](https://user-images.githubusercontent.com/17604849/231155158-7ec84454-d3fd-47a6-b849-33fc48896405.png)

Output:
![image](https://user-images.githubusercontent.com/17604849/231155208-aa90c1fb-cc03-4d21-a6cc-6d7c4c477a5f.png)

This app seems fake.

**Comment by kopyl** [comment]
No, it works, but in some strange unpredicted way. It would be good to have docs...

**Comment by kopyl** [comment]
Also:

![image](https://user-images.githubusercontent.com/17604849/231161596-703c8cc7-90e2-42ff-824f-a42c0388d6fc.png)

This is not a fork... How does it even work?

**Comment by GeorgePearse** [comment]
Hey @kopyl with the first. The model would still likely score more highly for 'human ear' than 'cat ear', and you can process that with something like NMS.

For the second, that's a pretty non-standard image, worth trying 'cartoon owl drawing' and seeing if it scores more highly than 'fork' though. It is always the case that models need to be tuned / thresholds need to be set for different classes. 

Fair point that the technique maybe struggles with 'false positives' but it is an openset detector, think of it more as trying to find examples of the text you've given it, than as a classifier. If you want better behaviour, give the model alternatives, would recommend @SkalskiP  's youtube tutorial https://www.youtube.com/watch?v=cMa77r3YrDk

**Comment by SkalskiP** [comment]
@GeorgePearse, thanks a lot for the referral 🙏

**Comment by kopyl** [comment]
@GeorgePearse thanks :))
